install/cupy_builder/cupy_setup_build.py

```diff
@@ -343,15 +343,23 @@ def make_extensions(ctx: Context, compiler, use_cython):
             # that are executable in the build environment, not the target
             # environment.
             compiler = ccompiler.new_compiler()
+            sysconfig.customize_compiler(compiler)
```

Comment:
In Conda we use this to pick up compiler details from the Python build. Seems CuPy uses this in most places except here. Likely this is because of the heavy patching to perform tests with the build platform compiler instead of host compiler, which is otherwise used

However without this, we get errors like `c++` not found, which can be a bit cryptic

Now if we get an error it points to the C++ compiler used (like the host one), which we will more easily identify as wrong and fix

Also there can be other details that are useful to fill in for the compiler here

Reply:
So does it mean adding this line alone is enough to fix the c++ not found error, and that the rest of the changes are mostly cosmetic?

Reply:
No, it means that this loads the executables targeting the `host_platform`. AIUI we should only use executables targeting the `build_platform`. So this makes it easier to see when we are missing something from the patch as references to the `host_platform` will give us a clue

Reply:
Thanks, John, good catch! I think the doc could have been more clear that a `new_compiler()` call should be always followed by `customize_compiler()`. I do see it's always done correctly in CuPy so far ([link](https://github.com/search?q=repo%3Acupy%2Fcupy+new_compiler&type=code)) except here.